app/ui/base/src/main/java/com/fsck/k9/ui/base/ThemeManager.kt

```diff
@@ -44,11 +56,17 @@ class ThemeManager(
     val translucentDialogThemeResourceId: Int = themeProvider.translucentDialogThemeResourceId
 
     fun init() {
-        updateAppTheme()
+        generalSettingsManager.getSettingsFlow()
+            .map { it.appTheme }
+            .distinctUntilChanged()
+            .onEach {
+                updateAppTheme(it)
+            }
+            .launchIn(appCoroutineScope)
```

Comment:
During testing the app crashed. `updateAppTheme()` needs to be called from the main thread. So this line needs to be…

```suggestion
            .launchIn(appCoroutineScope + Dispatchers.Main)
```